src/components/Menu.tsx

```diff
@@ -189,204 +189,220 @@ function DeleteFile(props: { item: MEMenuItem; close: () => void }) {
+                    Move
                   </Button>
-                ) : null}
-                {nfcRunning && props.item.type !== "folder" && props.item.type !== "zip" ? (
                   <Button
                     variant="outlined"
-                    sx={{mt: 1}}
-                    startIcon={<TapAndPlayIcon/>}
-                    onClick={() => {
-                      if (props.item.path) {
-                        setWaitingNfc(true);
-                        api.nfcWrite({
-                          path: props.item.path
-                        }).then(() => {
-                          setOpen(false);
-                        }).finally(() => {
-                          setWaitingNfc(false);
-                        });
-                      }
-                    }}
+                    startIcon={<DeleteForeverIcon/>}
+                    onClick={() => setEditMode(EditMode.Delete)}
+                    color="error"
                   >
-                    {waitingNfc ? "Waiting for tag..." : "Write to NFC tag"}
+                    Delete
                   </Button>
-                ) : null}
-                {!props.item.inZip ? (
-                  <>
-                    <Button
-                      variant="outlined"
-                      startIcon={<DriveFileRenameOutlineIcon/>}
-                      onClick={() => setEditMode(EditMode.Rename)}
-                    >
-                      Rename
-                    </Button>
-                    <Button
-                      variant="outlined"
-                      startIcon={<DriveFileMoveIcon/>}
-                      onClick={() => setEditMode(EditMode.Move)}
-                    >
-                      Move
-                    </Button>
-                    <Button
-                      variant="outlined"
-                      startIcon={<DeleteForeverIcon/>}
-                      onClick={() => setEditMode(EditMode.Delete)}
-                      color="error"
-                    >
-                      Delete
-                    </Button>
-                  </>
-                ) : null}
-              </Stack>
-            </DialogContent>
-          </Box>
-        );
+                </>
+              ) : null}
+            </Stack>
+          </DialogContent>
+        </Box>
+      );
     }
+}
+
+function EditFile(props: {
+  item: MEMenuItem;
+  parentContents: MEMenuItem[];
+  refresh: () => void;
+  setOpenShortcut?: (open: boolean) => void;
+  setSelectedPath?: (path: string) => void;
+}) {
+  const [open, setOpen] = useState<boolean>(false);
+
+  const handleClose = () => {
+    setOpen(false);
+    props.refresh();
   };
 
   return (
     <>
       <IconButton sx={{mr: -1}} onClick={() => setOpen(true)}>
         <MoreVertIcon/>
       </IconButton>
-      <Dialog open={open} onClose={handleClose}>
-        {editSection()}
-      </Dialog>
+      {open && (<Dialog open={open} onClose={handleClose}>
```

Comment:
2 things here:
- tuck away expensive hooks in EditSection
- be sure Dialog and EditSection aren't rendered when are not open